en/contribute/git_examples.md

```diff
@@ -72,30 +82,122 @@ For a good commit message, please refer to [Contributing](../contribute/README.m
   Responsible members of PX4 will now have a look at your contribution and decide if they want to integrate it.
   Check if they have questions on your changes every once in a while.
 
+
+## Changing Source Trees
+
+We recommend using PX4 `make` commands to switch between source code branches.
+This saves you having to remember the commands to update submodules and clean up build artifacts (build files that are not removed will result in "untracked files" errors after the switch).
+
+To switch between branches:
+
+1. Clean up the current branch, de-initializing submodule and removing all build artifacts:
+
+   ```sh
+   make clean
+   make distclean
+   ```
+1. Switch to a new branch or tag (here we first fetch the fictional branch "PR_test_branch" from the `upstream` remote):
+
+   ```sh
+   git fetch upstream PR_test_branch
+   git checkout PR_test_branch
+   ```
+1. Get the submodules for the new branch:
+
+   ```sh
+   make submodulesclean
+   ```
+
+<!-- FYI: Cleaning commands in https://github.com/PX4/PX4-Autopilot/blob/main/Makefile#L494 -->
+
+
 ## Get a Specific Release
 
-To get the source code for a *specific older release*:
-* Clone the PX4-Autopilot repo and navigate into PX4-Autopilot directory:
+Specific PX4 point releases are made as tags of the [release branches](#get-a-release-branch), and are named using the format `v<release>`.
+These are [listed on Github here](https://github.com/PX4/PX4-Autopilot/releases?q=release&expanded=true) (or you can query all tags using `git tag -l`).
+
+To get the source code for a *specific older release* (tag):
+
+1. Clone the PX4-Autopilot repo and navigate into _PX4-Autopilot_ directory:
+
+   ```sh
+   git clone https://github.com/PX4/PX4-Autopilot.git
+   cd PX4-Autopilot
+   ```
+   
+   :::note
+   You can reuse an existing repo rather than cloning a new one.
+   In this case clean the build environment (see [changing source trees](#changing-source-trees)):
+   
+   ```sh
+   make clean
+   make distclean
+   ```
+   :::
+
+1. Checkout code for particular tag (e.g. for tag v1.13.0-beta2)
+
+   ```sh
+   git checkout v1.13.0-beta2
+   ```
+
+1. Update submodules:
```

Comment:
FWIW it is probably either-or on `make submodulesclean` and `make distclean` - updating modules after switching branches seems to work find for me normally if I have done a distclean. I guess this builds good habits though.